src/components/UTIL_FormField.component

```diff
@@ -0,0 +1,113 @@
+    <apex:variable var="ftype" value="{!$ObjectType[sObjType].fields[field].type}"/>
+    <!-- The fixup style class to apply to dependent and date fields -->
+    <apex:variable var="fixup"
+        value="{!IF(dependent,
+                    IF(ftype=='picklist', ' dependentPicklistFixup', ' dependentMultiPicklistFixup'),
+                    IF(CONTAINS(ftype,'date'),' dateFixup', ''))}"/>
+    <!-- The field label to be displayed -->
+    <apex:variable var="label" value="{!$ObjectType[sObjType].fields[field].label}"/>
+
+    <!-- Read only fields will be displayed as outputfields with a bottom divider per slds standards, checkboxes will render as disabled input checkboxes -->
+    <apex:outputPanel layout="none" id="divOutput"
+        rendered="{!((!$ObjectType[sObjType].fields[field].updateable && sObj['Id']!='')
+                    || !$ObjectType[sObjType].fields[field].createable) && $ObjectType[sObjType].fields[field].accessible}">
+        <div class="{!'slds-form-element ' + styleClass}">
+            <apex:outputLabel for="outputX" styleClass="slds-form-element__label" rendered="{!ftype!='boolean'}">
+                {!label}
+            </apex:outputLabel>
+            <div class="{!'slds-form-element__control' + IF($ObjectType[sObjType].fields[field].type!='boolean', ' slds-has-divider--bottom slds-p-left--small','')}">
+                <!-- Blank space fixes issue where an empty span on read only fields were blank causing the line to shift when no value present -->
+                <apex:outputField id="outputX" value="{!sObj[field]}" styleClass="slds-form-element__static" rendered="{!ftype!='boolean'}">&nbsp;</apex:outputField>
+                <apex:outputPanel layout="{!IF(ftype=='boolean', 'inline', 'none')}" styleClass="{!IF(ftype=='boolean', 'slds-checkbox', '')}">
+                    <apex:outputLabel for="outputCB" styleClass="slds-checkbox__label" rendered="{!ftype=='boolean'}">
+                        <apex:inputCheckbox value="{!sObj[field]}" id="outputCB" disabled="true" rendered="{!ftype=='boolean'}" />
+                        <span class="slds-checkbox--faux"></span>
+                        <span class="slds-form-element__label">{!label}</span>
+                    </apex:outputLabel>
+                </apex:outputPanel>
+            </div>
+        </div>
+    </apex:outputPanel>
+
+    <!-- Updateable fields will render as input fields with the appropriate slds style - note that Record Type Id is displayed as a picklist by Salesforce. -->
+    <apex:outputPanel layout="block" styleClass="{!'slds-form-element ' + styleClass + fixup}" id="divInput"
+        rendered="{!($ObjectType[sObjType].fields[field].updateable
+                    || $ObjectType[sObjType].fields[field].createable && sObj['Id']=='') && $ObjectType[sObjType].fields[field].type != 'boolean'}">        
```

Comment:
Conditionally rendering output is where there is a bug and we have to use the full expression. There is a note up where the variable is declared.